content/en/docs/ambient/usage/extend-waypoint-wasm/index.md

```diff
@@ -0,0 +1,240 @@
+waypoint default/reviews-svc-waypoint applied
+{{< /text >}}
+
+Label the `reviews` service to use the `reviews-svc-waypoint` waypoint:
+
+{{< text bash >}}
+$ kubectl label service reviews istio.io/use-waypoint=reviews-svc-waypoint
+service/reviews labeled
+{{< /text >}}
+
+Any requests from pods in the ambient mode to the `reviews` service will now be routed through the `reviews-svc-waypoint` waypoint.
+
+{{< text bash >}}
+$ kubectl get gateway
+NAME                   CLASS            ADDRESS         PROGRAMMED   AGE
+bookinfo-gateway       istio            172.18.7.110    True         46h
+reviews-svc-waypoint   istio-waypoint   10.96.177.137   True         30s
+waypoint               istio-waypoint   10.96.202.82    True         44h
+{{< /text >}}
+
+Now, create a `WasmPlugin` targeting the new `waypoint` so that the extension applies only to the `reviews` service. In this configuration, the authentication token and the prefix are tailored specifically for the reviews service, ensuring that only requests directed towards it are subjected to this authentication mechanism.
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: basic-auth-for-service
+spec:
+  targetRefs:
+    - kind: Gateway
+      group: gateway.networking.k8s.io
+      name: reviews-svc-waypoint
+  url: oci://ghcr.io/istio-ecosystem/wasm-extensions/basic_auth:1.12.0
+  phase: AUTHN
+  pluginConfig:
+    basic_auth_rules:
+      - prefix: "/reviews"
+        request_methods:
+          - "GET"
+          - "POST"
+        credentials:
+          - "ok:test"
+          - "MXQtaW4zOmFkbWluMw=="
+EOF
+{{< /text >}}
+
+### Verify the traffic targeting the Service
+
+1. Test internal `/productpage` with credentials configured at the generic `waypoint` proxy
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null -H "Authorization: Basic YWRtaW4zOmFkbWluMw==" http://productpage:9080/productpage
+    200
+    {{< /text >}}
+
+1. Test internal `/reviews` with credentials configured at the specific `reviews-svc-waypoint` proxy
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null -H "Authorization: Basic MXQtaW4zOmFkbWluMw==" http://reviews:9080/reviews/1
+    200
+    {{< /text >}}
+
+1. Test internal `/productpage` without credentials
+
+    {{< text bash >}}
+    $ kubectl exec deploy/sleep -- curl -s -w "%{http_code}" -o /dev/null http://reviews:9080/reviews/1
+    401
+    {{< /text >}}
+
+When executing the provided command without credentials, it verifies that accessing the internal `/productpage` results in a 401 unauthorized response, demonstrating the expected behavior of failing to access the resource without proper authentication credentials.
+
+### Cleanup
+
+{{< text bash >}}
+$ kubectl delete wasmplugin basic-auth-at-gateway basic-auth-at-waypoint basic-auth-for-service
```

Comment:
Add a clean up for waypoint? and remove the use-waypoint label?

Reply:
@linsun : i think it is not needed, as at line 240 we refer to ambient cleanup document which includes steps to clean up which already includes these steps? https://preliminary.istio.io/latest/docs/ambient/getting-started/#uninstall

Reply:
Oh sounds good, thx for the clarification!

Reply:
please add some descriptive text to this section above the command (look at how it's done elsewhere if you need)